server/src/test/java/org/elasticsearch/cluster/coordination/Zen1IT.java

```diff
@@ -45,4 +66,202 @@ public void testZen1NodesJoiningZen2Cluster() {
         internalCluster().startNodes(randomIntBetween(1, 3), ZEN1_SETTINGS);
         createIndex("test");
     }
+
+    public void testMixedClusterFormation() throws Exception {
+        final int zen1NodeCount = randomIntBetween(1, 3);
+        final int zen2NodeCount = randomIntBetween(1, 3);
```

Comment:
this test fails when running with one Zen1 and one Zen2 node.

Reply:
It does indeed now that we reload our metadata. I'm not sure that's a case we want to test - it needs to be able to downgrade a Zen2 cluster back to a Zen1 master and then back up again, and I think that's a recipe for split-brain. WDYT?

Reply:
yeah, I also think we don't need to test that situation.

Reply:
Ok I avoid testing the 2-node case here.